resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsWatchSnapshotSource.scala

```diff
@@ -86,8 +97,14 @@ class ExecutorPodsWatchSnapshotSource(
     }
 
     override def onClose(e: WatcherException): Unit = {
-      logWarning("Kubernetes client has been closed (this is expected if the application is" +
-        " shutting down.)", e)
+      if (e != null && e.isHttpGone) {
```

Comment:
I'm _slightly_ worried about an infinite loop here, I don't see any clear doc on when isHttpGone is set true in fabric8, can we have a max resets in K minutes counter or something similar?

Or if there is good docs on the behaviour of isHttpGone that would be good too.

Reply:
I think we can accept the isHttpGone check as kubernetes-client does the exact same here:

https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/impl/cache/Reflector.java#L288-L294